libyara/modules/elf/elf.c

```diff
@@ -102,7 +102,7 @@ define_function(telfhash)
         string functions (str.* and mem.*), gcc changes them depending on arch
         symbols starting with . or _ */
     bool is_bad_prefix = name[0] == '.' || name[0] == '_';
-    bool is_x86_64 = name[0] == '.' || name[0] == '_';
+    bool is_x86_64 = strncmp(name + strlen(name) - 2, "64", 2) == 0;
```

Comment:
I don't think a symbol name must be non-empty, so `strlen(name)` can return 0 and read into `name - 2` - pre-running the buffer.

Reply:
I think a non-empty symbol is impossible given we filtered symbols that are global and functions previously, however a one letter symbol is definitely possible, that's a mistake on my part, thanks.

I added a test using custom made elf file with a bit of everything to check the telfhash is correct. I added a one letter symbol and valgrind does indicate out of bound read, as expected.